sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/AdaptiveSparkPlanExec.scala

```diff
@@ -222,7 +222,7 @@ case class AdaptiveSparkPlanExec(
       .map(_.toLong).filter(SQLExecution.getQueryExecution(_) eq context.qe)
   }
 
-  private def getFinalPhysicalPlan(): SparkPlan = lock.synchronized {
+  private[sql] def getFinalPhysicalPlan(): SparkPlan = lock.synchronized {
```

Comment:
probably just remove `private`, as other Spark vendors may hit the same problem and need to materialize the AQE plan manually.